bookwyrm/templates/discover/small-book.html

```diff
@@ -1,12 +1,22 @@
+
 {% load bookwyrm_tags %}
 {% load i18n %}
+
 {% if book %}
-<a href="{{ book.local_path }}">{% include 'snippets/book_cover.html' with book=book %}</a>
-{% include 'snippets/stars.html' with rating=book|rating:request.user %}
+    {% with book=book %}
+        <a href="{{ book.local_path }}">{% include 'snippets/book_cover.html' %}</a>
 
-<h3 class="title is-6"><a href="/book/{{ book.id }}">{{ book.title }}</a></h3>
-{% if book.authors %}
-<p class="subtitle is-6">{% trans "by" %} {% include 'snippets/authors.html' with book=book %}</p>
```

Comment:
I realize this is a WIP, so feel free to disregard if this isn't helpful context -- the only reason I'm specifying `book=book` here is to be super explicit about the scope variables passed into the snippet. it would work the same without stating that, since the scope is inherited from the parent template. So adding `{% with book=book %}` isn't necessary either - it's just a matter of if you think it makes it clearer

Reply:
Thinking about it, would it be helpful to have a global `with` wrap listing the variables that are expected to be available in the template?

Would there be adverse consequences?

Reply:
Thinking about it, would it be helpful to have a global `with` wrap listing the variables that are expected to be available in the template? A sort of documentation… :)

Would there be adverse consequences?

Reply:
I'd prefer using comments to document rather than variable assignments, I think

Reply:
Sure. :)

I’ll try to move those `with` variables in a comment at the beginning of each template from now on. That will also reduce the number of indentations.